### ISSUE TITLE
Prioritizing visible tables in SQL auto-suggest

### ISSUE THREAD

**Issue by robinfrick** [metabase/metabase#2230]
This is a UX proposal: When typing a SQL query Metabase suggests tables to choose from.
As mentioned before (when I asked to be able to hide/unhide schemas) I have a lot of tables in schemas that metabase doesn't even have access to. I hide these tables. Those tables appear as top suggestions when typing a query in SQL. I think it makes sense that they are suggested as I can query them through SQL, but I'd like to see tables that are designated as queryable in the admin panel being prioritized as it is more likely that these are the actually tables I want.

**Comment by mazameli** [comment]
Man, this makes a ton of sense. Great catch, @robinfrick

**Comment by salsakran** [comment]
@mazameli do you think would we need any visual elements or would this simply change the order?

**Comment by mazameli** [comment]
Simply changing the order would suffice, bonus points for putting an `<optgroup>` type heading above the hidden table(s). Like this:

```
<select>
    <option value="table1">Table 1</option>
    <option value="table2">Table 2</option>
  <optgroup label="Hidden Tables">
    <option value="htable1">Hidden Table 1</option>
    <option value="htable2">Hidden Table 2</option>
  </optgroup>
</select>
```

**Comment by robinfrick** [comment]
I'll add to that: Please completely hide suggestions from all database but the one that is selected for this query. I get suggestions from our postgres database while having redshift selected as the database for the query I'm writing.

**Comment by salsakran** [comment]
@robinfrick yeah, I'd classify that as an outright bug. Can you try to reproduce it and open a bug with browser/version info? I tried to repro it on our staging environment and wasn't able to get it to happen.

**Comment by robinfrick** [comment]
@salsakran luckily I can't reproduce it. So let's just say I was tired an seeing things that weren't supposed to be. If I start seeing things again I'll try to see if there's a pattern besides long work hours and tell you.

**Comment by flamber** [comment]
Closing, since hidden tables has not been included in the auto-suggest since 0.36.0
